### PR TITLE
WIP: add webpack5 support

### PR DESCRIPTION
**Pull request by nmalyschkin**
- [ ] revisit all README and package.json file and make sure there are no copy mistakes
- [ ] consider unifying the `helpers.ts` files to reduce code duplication

Why are these tools not part of the piral monorepo?

### CODE REVIEW

**Review by FlorianRappl — COMMENTED** [review]
In general is great.

However, (just to be on the same page as in the piral repo PR) we should reflect it we really want these as new plugins. Also regarding the name I am not sure if `enchancer` was meant or if its a typo for `enhancer`.

File: packages/pilet-webpack-config-enchancer/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        banner: `//@pilet v:${bannerSuffix}`,
+        entryOnly: true,
+        include: /\.js$/,
+        raw: true,
+      }),
+    );
+  }
+
+  // FIXME: this is a really bad idea, making finding the source of configuration nearly impossible
+  setEnvironment(variables);
```
Comment (FlorianRappl) [inline packages/pilet-webpack-config-enchancer/src/index.ts:74]: Do we need this for webpack5 ? The `process.env` is not used anyway. The only thing I can think of is the `NODE_ENV`, but here, too, its more about the `DefinePlugin` IIRC.

File: packages/pilet-webpack-config-enchancer/package.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    "node": ">= 8"
+  },
+  "keywords": [
+    "piral-cli",
+    "piral",
+    "plugin",
+    "webpack",
+    "command"
+  ],
+  "author": "Florian Rappl <mail@florian-rappl.de>",
```
Comment (FlorianRappl) [inline packages/pilet-webpack-config-enchancer/package.json:25]: Feel free to have your name in these new packages.

File: packages/pilet-webpack-config-enchancer/README.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+Allows supplying additional variables to be used as definitions. Similar to the `definePlugin`.
+
+Example:
+
+```js
+const { piletWebpackConfigEnchancer } = require("pilet-webpack-config-enchancer");
+const piletPkg = require("./package.json");
+
+const enchance = piletWebpackConfigEnchancer({
```
Comment (FlorianRappl) [inline packages/pilet-webpack-config-enchancer/README.md:57]: Love the documentation and the curried approach here.

**Comment by nmalyschkin** [comment]
https://github.com/smapiot/piral/pull/334#issuecomment-727802189

It was decided to put the enhancers directly in the `piral-cli-webpack5` package, making this PR obsolete.

**Pull request closed without merging**